### PR TITLE
tcp listener: log source IP

### PR DESCRIPTION
**Pull request by rgs1**
Useful while debugging TLS (and general) connection issues.

Signed-off-by: Raul Gutierrez Segales <rgs@pinterest.com>

### CODE REVIEW

**Review by lambdai — DISMISSED** [review]
LGTM
Since the goal is debugging, you may want to add the source ip to 
`ENVOY_LOG(debug, "closing connection: no matching filter chain found");`
and
`ENVOY_CONN_LOG(debug, "closing connection: no filters", *active_connection->connection_);`

It's up to you.

File: source/server/active_tcp_listener.cc [hunk 1]
```diff
@@ -325,7 +325,8 @@ void ActiveTcpListener::newConnection(Network::ConnectionSocketPtr&& socket,
 
   // If the connection is already closed, we can just let this connection immediately die.
   if (active_connection->connection_->state() != Network::Connection::State::Closed) {
-    ENVOY_CONN_LOG(debug, "new connection", *active_connection->connection_);
+    ENVOY_CONN_LOG(debug, "new connection from {}", *active_connection->connection_,
+                   active_connection->connection_->addressProvider().remoteAddress()->asString());
```
Comment (zuercher) [inline source/server/active_tcp_listener.cc:329]: Do we need to guard against a nullptr return from `remoteAddress`? It seems generally possible, but maybe not on any code path that end up here.

Comment (rgs1) [reply source/server/active_tcp_listener.cc:329]: I think we are good for this path. Shall I add an assert at least?

Comment (zuercher) [reply source/server/active_tcp_listener.cc:332]: Yeah, let's do that.

File: source/server/active_tcp_listener.cc [hunk 2]
```diff
@@ -311,21 +314,27 @@ void ActiveTcpListener::newConnection(Network::ConnectionSocketPtr&& socket,
...
 
+  ASSERT(active_connection->connection_->addressProvider().remoteAddress() != nullptr);
+
   const bool empty_filter_chain = !config_->filterChainFactory().createNetworkFilterChain(
       *active_connection->connection_, filter_chain->networkFilterFactories());
   if (empty_filter_chain) {
-    ENVOY_CONN_LOG(debug, "closing connection: no filters", *active_connection->connection_);
+    ENVOY_CONN_LOG(debug, "closing connection from {}: no filters",
+                   active_connection->connection_->addressProvider().remoteAddress()->asString(),
+                   *active_connection->connection_);
```
Comment (yanavlasov) [inline source/server/active_tcp_listener.cc:330]: This needs to be the first parameter.

**Review by yanavlasov — CHANGES_REQUESTED** [review]

File: source/server/active_tcp_listener.cc (hunk 2 above)
Comment (rgs1) [reply source/server/active_tcp_listener.cc:330]: ah right, sorry

File: source/server/active_tcp_listener.cc [hunk 3]
```diff
@@ -214,8 +214,9 @@ void ActiveTcpSocket::newConnection() {
 
 void ActiveTcpListener::onAccept(Network::ConnectionSocketPtr&& socket) {
   if (listenerConnectionLimitReached()) {
-    ENVOY_LOG(trace, "closing connection: listener connection limit reached for {}",
-              config_->name());
+    ASSERT(socket->addressProvider().remoteAddress() != nullptr);
```
Comment (zuercher) [inline source/server/active_tcp_listener.cc:217]: I should have checked this earlier, but per STYLE.md this should be a RELEASE_ASSERT, I think. It explicitly calls out this case as somewhere we shouldn't just ASSERT.

**Review by zuercher — APPROVED** [review]
lgtm @yanavlasov do you want to do a final pass as well?

**Review by yanavlasov — APPROVED** [review]

File: source/server/active_tcp_listener.cc (hunk 3 above)
Comment (mattklein123) [reply source/server/active_tcp_listener.cc]: A drive by: As this case is going to crash in an obvious way on the next line I think in general we probably don't need any assert here, but if we want to document our assumption I think a normal ASSERT is fine here. cc @zuercher @rgs1. I might be misunderstanding the new style guide though.

**Pull request merged**